python/pyarrow/parquet.py

```diff
@@ -966,6 +966,14 @@ def write_table(table, where, row_group_size=None, version='1.0',
 """.format(_parquet_writer_arg_docs)
 
 
+def _mkdir_if_not_exists(fs, path):
+    if fs._isfilestore() and not fs.exists(path):
+        try:
+            fs.mkdir(path)
+        except OSError:
+            assert fs.exists(path)
```

Comment:
It might be worth adding `exist_ok` to the filesystem interface. Python has this kwarg for `os.makedirs` (but not `os.mkdir`) to say not to error if the directory already exists (defaults to False).

Reply:
Before adding more things to the interface, I would prefer that we continue to settle the discussion on a uniform file interface between `dask`, Arrow and others. Then we actually know to what we all want to converge to.

Reply:
Maybe worth opening a JIRA about this in either case